tests_e2e/tests/agent_status/agent_status.py

```diff
@@ -0,0 +1,192 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# Validates the agent status is updated without processing additional goal states (aside from the first goal state
+# from fabric)
+#
+
+from azure.mgmt.compute.models import VirtualMachineInstanceView, InstanceViewStatus, VirtualMachineAgentInstanceView
+from assertpy import assert_that
+from datetime import datetime, timedelta
+from time import sleep
+import json
+
+from tests_e2e.tests.lib.agent_test import AgentTest
+from tests_e2e.tests.lib.agent_test_context import AgentTestContext
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.virtual_machine_client import VirtualMachineClient
+
+
+class RetryableAgentStatusException(BaseException):
+    pass
+
+
+class AgentStatus(AgentTest):
+    def __init__(self, context: AgentTestContext):
+        super().__init__(context)
+        self._ssh_client = self._context.create_ssh_client()
+
+    def validate_instance_view_vmagent_status(self, instance_view: VirtualMachineInstanceView):
+        status: InstanceViewStatus = instance_view.vm_agent.statuses[0]
+
+        # Validate message field
+        if status.message is None:
+            raise RetryableAgentStatusException("Agent status is invalid: 'message' property in instance view is None")
+        elif 'unresponsive' in status.message:
+            raise RetryableAgentStatusException("Agent status is invalid: Instance view shows unresponsive agent")
+
+        # Validate display status field
+        if status.display_status is None:
+            raise RetryableAgentStatusException("Agent status is invalid: 'display_status' property in instance view is None")
+        elif 'Not Ready' in status.display_status:
+            raise RetryableAgentStatusException("Agent status is invalid: Instance view shows agent status is not ready")
+
+        # Validate time field
+        if status.time is None:
+            raise RetryableAgentStatusException("Agent status is invalid: 'time' property in instance view is None")
+
+    def validate_instance_view_vmagent(self, instance_view: VirtualMachineInstanceView):
+        """
+        Checks that instance view has vm_agent.statuses and vm_agent.vm_agent_version properties which report the Guest
+        Agent as running and Ready:
+
+        "vm_agent": {
+            "extension_handlers": [],
+            "vm_agent_version": "9.9.9.9",
+            "statuses": [
+                {
+                    "level": "Info",
+                    "time": "2023-08-11T09:13:01.000Z",
+                    "message": "Guest Agent is running",
+                    "code": "ProvisioningState/succeeded",
+                    "display_status": "Ready"
+                }
+            ]
+        }
+        """
+        if instance_view.vm_agent is None:
+            raise RetryableAgentStatusException("Agent status is invalid: 'vm_agent' property in instance view is None")
```

Comment:
do you think these properties will ever be None? I think more likely they would be missing, as your previous code was trying to check